salt/log/setup.py

```diff
@@ -1088,6 +1262,70 @@ def patch_python_logging_handlers():
         logging.handlers.QueueHandler = QueueHandler
 
 
+def __process_multiprocessing_logging_zmq(opts, port):
+    # Avoid circular import
+    import salt.utils.process
+    salt.utils.process.appendproctitle('MultiprocessingLoggingQueue')
+
+    # Assign UID/GID of user to proc if set
+    from salt.utils.verify import check_user
+    user = opts.get('user')
+    if user:
+        check_user(user)
+
+    from salt.utils.platform import is_windows
+    if is_windows():
+        # On Windows, creating a new process doesn't fork (copy the parent
+        # process image). Due to this, we need to setup all of our logging
+        # inside this process.
+        setup_temp_logger()
+        setup_console_logger(
+            log_level=opts.get('log_level'),
+            log_format=opts.get('log_fmt_console'),
+            date_format=opts.get('log_datefmt_console')
+        )
+        setup_logfile_logger(
+            opts.get('log_file'),
+            log_level=opts.get('log_level_logfile'),
+            log_format=opts.get('log_fmt_logfile'),
+            date_format=opts.get('log_datefmt_logfile'),
+            max_bytes=opts.get('log_rotate_max_bytes', 0),
+            backup_count=opts.get('log_rotate_backup_count', 0)
+        )
+        setup_extended_logging(opts)
+
+    import traceback
+    context = zmq.Context()
+    puller = context.socket(zmq.PULL)
+    try:
+        puller.bind('tcp://127.0.0.1:{}'.format(port))
+    except Exception:
+        print("MP LOGGING TB %s" %('\n'.join(traceback.format_stack())))
+        raise
+    try:
+        while True:
+            try:
+                record_dict = msgpack.loads(puller.recv())
```

Comment:
Why not the msgpack unpacker approach?
Because we always get full payloads?

Reply:
Yes

Reply:
Ok